cdap-common/bin/functions.sh

```diff
@@ -503,15 +503,31 @@ cdap_set_spark() {
       # If there is no valid SPARK_HOME, we should unset the existing one if it is set
       # Otherwise the spark-shell won't run correctly
       unset SPARK_HOME
+      local __spark_shell=$(which spark-shell)
+      local __spark_client_version=None
+      for __dist in hdp iop; do
+        if [[ $(which ${__dist}-select 2>/dev/null) ]]; then
+          __spark_client_version=$(${__dist}-select status spark-client | awk '{print $3}')
+        fi
+        if [[ ${__spark_client_version} == 'None' ]]; then # defaults None, we're hoping for a version
+          logecho "$(date) Spark client not installed via ${__dist}-select detection"
+          return 1
```

Comment:
how does this work on iop?  Won't the first loop iteration, testing for `hdp`, hit this and return prematurely?